Source/pfile.cpp

```diff
@@ -59,43 +59,34 @@ void pfile_encode_hero(const PkPlayerStruct *pPack)
 
 BOOL pfile_open_archive(BOOL update, DWORD save_num)
 {
-	char FileName[MAX_PATH];
-
-	pfile_get_save_path(FileName, sizeof(FileName), save_num);
-	if (OpenMPQ(FileName, FALSE, save_num))
+	const std::string FileName = pfile_get_save_path(save_num);
+	if (OpenMPQ(FileName.c_str(), FALSE, save_num))
 		return TRUE;
 
 	if (update && gbMaxPlayers > 1)
 		mpqapi_store_default_time(save_num);
 	return FALSE;
 }
 
-void pfile_get_save_path(char *pszBuf, DWORD dwBufSize, DWORD save_num)
+std::string pfile_get_save_path(DWORD save_num)
 {
-	char path[MAX_PATH];
-
+	std::string result = GetPrefPath();
 #ifdef SPAWN
-	const char *fmt = "%sshare_%d.sv";
-
-	if (gbMaxPlayers <= 1)
-		fmt = "%sspawn%d.sv";
+	result.append(gbMaxPlayers <= 1 ? "share_" : "spawn");
 #else
-	const char *fmt = "%smulti_%d.sv";
-
-	if (gbMaxPlayers <= 1)
-		fmt = "%ssingle_%d.sv";
+	result.append(gbMaxPlayers <= 1 ? "single_" : "multi_");
 #endif
-
-	GetPrefPath(path, MAX_PATH);
-	snprintf(pszBuf, MAX_PATH, fmt, path, save_num);
+	char save_num_str[8];

```

Comment:
Afaik save_num is limited to 0-9, so 2 chars would be enough.

Reply:
Acknowledged, I think we may want to allow more saves in the future (and this is unlikely to be tested).